### ISSUE TITLE
Nucleus instance segmentation classification performance very poor

### ISSUE THREAD

**Issue by deeplearningmaniac** [TissueImageAnalytics/tiatoolbox#756]
- TIA Toolbox version: 1.4.1
- Python version: 3.10.12
- Operating System: Ubuntu 22.04.3 LTS

### Description

I've been trying to use some of the pretrained nucleus instance segmentation models to segment, in particular, epithelial cells and lymphocytes. However, from some of my testing, all of the models perform very poorly. I was hoping you might be able to help me figure out if I'm going wrong somewhere.

### What I Did

I downloaded some data from the [NuCLS](https://sites.google.com/view/nucls/home) dataset, ran the different pretrained models on some patches, and visually compared the results with the ground truth. All three models I tried -- PanNuke, MoNuSAC, and CoNSeP -- were able to segment the nuclei to varying degrees, but none of them were able to properly classify any epithelial cells, and MoNuSAC wasn't able to classify lymphocytes either.

Here is a Google Colab notebook replicating this: https://colab.research.google.com/drive/1D3h4A2R7LvMUhUbel8YullU3gkL4AVbm?usp=sharing

Am I doing something wrong? I've checked my code over and over again and don't see any obvious bugs. Any help would be appreciated.

If you'd like to try some patches other than the two that I included with the Google Drive download (which I custom uploaded for simplicity), you can find them here: https://drive.google.com/drive/folders/1eGlF9Dgu3WMEik4fqj0wJ13LKVufsfZ0 under the `rgb/` and `csv/` folders.

**Comment by shaneahmed** [comment]
Thanks @deeplearningmaniac, we will investigate this. It should work with PanNuke. There is a model trained on much larger dataset i.e., Conic that should help for now https://github.com/TissueImageAnalytics/CoNIC/blob/main/notebooks/baseline.ipynb

**Comment by Taylor-Kelce** [comment]
I noticed that hovernet from `consep` and `panuke` give very different classification. For example, I used sample image provided by example notebook 08 with the two different hovernets: majority of the purple coloured nuclei are labeled as neoplastic epithelial by ![Pannuke](https://github.com/TissueImageAnalytics/tiatoolbox/assets/153136699/e2b82ba2-05ce-4cb4-8e0f-6826ad82a718) while they are labeled as background with HoverNet trained from ![Consep](https://github.com/TissueImageAnalytics/tiatoolbox/assets/153136699/59a8de66-f41f-4154-ade7-c1437512bbe7).

**Comment by shaneahmed** [comment]
Please note that models trained on different datasets will produce different results. We do not claim that the models we have trained would be flawless but these models should be able to produce results as listed in the documentation https://tia-toolbox.readthedocs.io/en/latest/_autosummary/tiatoolbox.models.architecture.hovernet.HoVerNet.html. 

The best model in our opinion, so far, is the one provided with the CONIC challenge https://github.com/TissueImageAnalytics/CoNIC/tree/main/notebooks. You can directly load the model from CONIC using the notebooks provided. 

We are also further clarifying the classes for the models in #751

**Comment by shaneahmed** [comment]
> I noticed that hovernet from `consep` and `panuke` give very different classification. For example, I used sample image provided by example notebook 08 with the two different hovernets. The majority of the purple coloured nuclei are labeled as neoplastic epithelial by Pannuake: ![Pannuke](https://private-user-images.githubusercontent.com/153136699/292710177-e2b82ba2-05ce-4cb4-8e0f-6826ad82a718.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUwNzAxNjksIm5iZiI6MTcwNTA2OTg2OSwicGF0aCI6Ii8xNTMxMzY2OTkvMjkyNzEwMTc3LWUyYjgyYmEyLTA1Y2UtNGNiNC04ZTBmLTY4MjZhZDgyYTcxOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExMlQxNDMxMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wYzA4YTQxODNmODZmZDk1YzdjMzk4YmZmMDBjMDIzOTEzYmYyZTQ1OGE4MDQxM2Q2YmQ2OWNjYWI0NmFhMzk4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.3DNd9C-aVXkxS-jXr3A7ne9MREv6_0hdoCXtF8n4LHA)
> 
> On the other hand, they are labeled as background with HoverNet trained from Consep: ![Consep](https://private-user-images.githubusercontent.com/153136699/292710156-59a8de66-f41f-4154-ade7-c1437512bbe7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUwNzAxNjksIm5iZiI6MTcwNTA2OTg2OSwicGF0aCI6Ii8xNTMxMzY2OTkvMjkyNzEwMTU2LTU5YThkZTY2LWY0MWYtNDE1NC1hZGU3LWMxNDM3NTEyYmJlNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTEyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExMlQxNDMxMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNDVmZmYxOWIwZWNmYmE4MDI1YzMzMDA5YmY2Yzg0YzJjOWNmNzI0MWRjZjdlNzQ0ZmU3MzkwOGJjZWE4YzYzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.EX2wRoPHdgJxEs-mnxA1aimHVfQ04IObAf-j02j3Jps).

The classification labels might be incorrect in this case. We are adding the documentation in #751
